pkg/hubble/filters/event_type.go

```diff
@@ -19,24 +19,34 @@ import (
 
 	flowpb "github.com/cilium/cilium/api/v1/flow"
 	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
 )
 
 func filterByEventType(types []*flowpb.EventTypeFilter) FilterFunc {
 	return func(ev *v1.Event) bool {
-		event := ev.GetFlow().GetEventType()
-		if event == nil {
-			return false
-		}
-
-		for _, typeFilter := range types {
-			if t := typeFilter.GetType(); t != 0 && event.Type != t {
-				continue
+		switch ev.Event.(type) {
+		case *flowpb.Flow:
+			event := ev.GetFlow().GetEventType()
+			for _, typeFilter := range types {
+				if t := typeFilter.GetType(); t != 0 && event.Type != t {
+					continue
+				}
+				if typeFilter.GetMatchSubType() && typeFilter.GetSubType() != event.SubType {
+					continue
+				}
+				return true
 			}
-
-			if typeFilter.GetMatchSubType() && typeFilter.GetSubType() != event.SubType {
-				continue
+		case *flowpb.AgentEvent:
```

Comment:
nit: Could you maybe add/extend a simple unit test ensuring that this case works as expected?

Reply:
Added `TestEventTypeFilter` which should cover the basic cases.